### DISCUSSION TITLE
Working with forms in QuestPDF

### DISCUSSION THREAD

**Discussion by elnikoff-t** [QuestPDF/QuestPDF#392] [Q&A] [answered]
All the question is there : 
I have a PDF Form, edited by an official entity.
This form contains a lot of fields, but many of them are unnamed or even named following text content (for example : there are checkboxes and their name are the full length text available for this checkbox).
I would like, in a first step, to be able to rename fields to meaningful names. 
And in a second step, I'd like to populate the form with data coming from another source.

I've made a lot of searchs, but I'd like to stick to a single developement environment. QuestPDF seems to be really promising, but... how to do all this work ? I'm totally new in .NET, while having lot of programming experience (mainly mainframes :s)

Thanks for all

**Comment by girlpunk** [comment, marked as answer]
Hi,

Unfortunately QuestPDF does not currently support modifying existing PDFs, only generating new ones.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

It is worth noting that QuestPDF now supports working with existing PDF files through the `DocumentOperation` API (merging, page selection, overlaying, and more), though form field editing remains outside the library's scope at this time.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
